internal/pkg/v2/controller/http/core/data/event.go

```diff
@@ -0,0 +1,81 @@
+package data
+
+import (
+	"net/http"
+
+	v2container "github.com/edgexfoundry/edgex-go/internal/pkg/v2/bootstrap/container"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/controller/errorconcept"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/handler/common"
+	handler "github.com/edgexfoundry/edgex-go/internal/pkg/v2/handler/core/data"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	commonDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	requestDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+	respDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/responses"
+)
+
+func EventController(
+	w http.ResponseWriter,
+	r *http.Request) {
+
+	if r.Body != nil {
+		defer func() { _ = r.Body.Close() }()
+	}
+
+	// retrieve all the registry clients
+	httpErrorHandler := v2container.RegistryClients.ErrorHandlerClient
+	lc := v2container.RegistryClients.LoggingClient
+	dbClient := v2container.RegistryClients.DBClient
+	chEvents := v2container.RegistryClients.EventChan
+	msgClient := v2container.RegistryClients.MessageClient
+	mdc := v2container.RegistryClients.MetadataClient
+	configuration := v2container.RegistryClients.ConfigClient
+
+	ctx := r.Context()
+
+	reader := common.NewRequestReader(r, configuration)
+	addEventReqDTOs, err := reader.Read(r.Body, &ctx)
+	if err != nil {
+		httpErrorHandler.HttpHandleOneVariant(w, err, errorconcept.NewErrContractInvalidError(err), errorconcept.Default.InternalServerError)
+		return
+	}
+	events := requestDTO.AddEventReqToEventModels(addEventReqDTOs)
+
+	// map Event models to AddEventResponse DTOs
+	var addResponses []respDTO.AddEventResponse
+	for i, e := range events {
+		newId, err := handler.AddNewEvent(e, ctx, lc, dbClient, chEvents, msgClient, mdc, configuration)
```

Comment:
You don't have to pass so many argument to the handler.  Most of them can be retrieve on its own.

Reply:
updated.